src/assert.rs

```diff
@@ -344,42 +347,51 @@ impl Assert {
 
         let mut spawned = command
             .spawn()
-            .chain_err(|| ErrorKind::SpawnFailed(self.cmd.clone()))?;
+            .chain_with(|| AssertionError::new(self.cmd.clone()))?;
 
         if let Some(ref contents) = self.stdin_contents {
             spawned
                 .stdin
                 .as_mut()
                 .expect("Couldn't get mut ref to command stdin")
-                .write_all(contents)?;
+                .write_all(contents)
+                .chain_with(|| AssertionError::new(self.cmd.clone()))?;
         }
-        let output = spawned.wait_with_output()?;
+        let output = spawned
+            .wait_with_output()
+            .chain_with(|| AssertionError::new(self.cmd.clone()))?;
 
         if let Some(expect_success) = self.expect_success {
-            if expect_success != output.status.success() {
-                let out = String::from_utf8_lossy(&output.stdout).to_string();
-                let err = String::from_utf8_lossy(&output.stderr).to_string();
-                let err: Error = ErrorKind::StatusMismatch(expect_success, out, err).into();
-                bail!(err.chain_err(|| ErrorKind::AssertionFailed(self.cmd.clone())));
+            let actual_success = output.status.success();
+            if expect_success != actual_success {
+                return Err(
+                    AssertionError::new(self.cmd.clone()).chain(StatusError::new(
```

Comment:
Creating the type of error that the return type expects is a little more ergonomic than it was under `failure::Error` which needs the `throw!` macro to be nice.